nacos/src/main/java/com/linecorp/armeria/client/nacos/NacosEndpointGroupBuilder.java

```diff
@@ -0,0 +1,183 @@
+     */
+    public NacosEndpointGroupBuilder registryFetchInterval(Duration registryFetchInterval) {
+        requireNonNull(registryFetchInterval, "registryFetchInterval");
+        checkArgument(!registryFetchInterval.isZero() && !registryFetchInterval.isNegative(),
+                "registryFetchInterval: %s (expected: > 0)",
+                registryFetchInterval);
+        return registryFetchIntervalMillis(registryFetchInterval.toMillis());
+    }
+
+    /**
+     * Sets the interval between fetching registry requests.
+     * If not set, {@value #DEFAULT_CHECK_INTERVAL_MILLIS} milliseconds is used by default.
+     */
+    public NacosEndpointGroupBuilder registryFetchIntervalMillis(long registryFetchIntervalMillis) {
+        checkArgument(registryFetchIntervalMillis > 0, "registryFetchIntervalMillis: %s (expected: > 0)",
+                registryFetchIntervalMillis);
+        this.registryFetchIntervalMillis = registryFetchIntervalMillis;
+        return this;
+    }
+
+    /**
+     * Sets the username and password pair for Nacos's API.
+     * Please refer to the
+     * <a href=https://nacos.io/en-us/docs/v2/guide/user/auth.html>Nacos Authentication Document</a>
+     * for more details.
+     *
+     * @param username the username for access Nacos API, default: {@code null}
+     * @param password the password for access Nacos API, default: {@code null}
+     */
+    public NacosEndpointGroupBuilder authorization(String username, String password) {
+        nacosClientBuilder.authorization(username, password);
+        return this;
+    }
+
+    /**
+     * Returns a newly-created {@link NacosEndpointGroup}.
+     */
+    public NacosEndpointGroup build() {
+        return new NacosEndpointGroup(selectionStrategy, shouldAllowEmptyEndpoints(), selectionTimeoutMillis(),
+                nacosClientBuilder.build(), registryFetchIntervalMillis);
+    }
+
+    @Override
+    public NacosEndpointGroupBuilder allowEmptyEndpoints(boolean allowEmptyEndpoints) {
+        return (NacosEndpointGroupBuilder) super.allowEmptyEndpoints(allowEmptyEndpoints);
+    }
+
+    /**
+     * Sets the timeout to wait until a successful {@link Endpoint} selection.
+     * {@link Duration#ZERO} disables the timeout.
+     * If unspecified, {@link Flags#defaultResponseTimeoutMillis()} is used by default.
+     */
+    @Override
+    public NacosEndpointGroupBuilder selectionTimeout(Duration selectionTimeout) {
+        return (NacosEndpointGroupBuilder) super.selectionTimeout(selectionTimeout);
+    }
+
+    /**
+     * Sets the timeout to wait until a successful {@link Endpoint} selection.
+     * {@code 0} disables the timeout.
+     * If unspecified, {@link Flags#defaultResponseTimeoutMillis()} is used by default.
+     */
+    @Override
+    public NacosEndpointGroupBuilder selectionTimeoutMillis(long selectionTimeoutMillis) {
+        return (NacosEndpointGroupBuilder) super.selectionTimeoutMillis(selectionTimeoutMillis);
+    }
```

Comment:
This is unrelated to your PR, but we can probably remove these lines now

Reply:
@jrhee17 
I have removed the code from `NacosEndpointGroupBuilder`. Should I also apply this to other extended classes of `AbstractDynamicEndpointGroupBuilder` on this PR?